build-logic-commons/code-quality/src/main/kotlin/gradlebuild.code-quality.gradle.kts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import gradlebuild.classycle.tasks.Classycle
+import gradlebuild.classycle.extension.ClassycleExtension
+
+plugins {
+    id("base")
+    id("checkstyle")
+    id("codenarc")
+}
+
+val codeQuality = tasks.register("codeQuality") {
+    dependsOn(tasks.matching { it is CodeNarc || it is Checkstyle || it is Classycle || it is ValidatePlugins })
+}
+tasks.withType<Test>().configureEach {
+    shouldRunAfter(codeQuality)
+}
+tasks.check {
+    dependsOn(codeQuality)
+}
+
+val rules by configurations.creating {
+    isVisible = false
+    isCanBeConsumed = false
+
+    attributes {
+        attribute(LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE, objects.named(LibraryElements.RESOURCES))
+    }
+}
+
+val classycle by configurations.creating {
+    isVisible = false
+    isCanBeConsumed = false
+
+    attributes {
+        attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage.JAVA_RUNTIME))
+    }
+}
+
+dependencies {
+    rules("gradlebuild:code-quality-rules") {
+        because("Provides rules defined in XML files")
+    }
+    codenarc("gradlebuild:code-quality-rules") {
+        because("Provides the IntegrationTestFixturesRule implementation")
+    }
+    codenarc("org.codenarc:CodeNarc:1.5")
```

Comment:
Should this version be extracted?

Reply:
With extraction you mean defined in a central place with the other library versions?

For that we need to pull out the "platforms" (potentially catalogs in the future) into a separate build which we can then use in all other included builds. We should do that at some point but there are some point. But I don't want to mix this into this PR. And there is only this one dependency in 'build-logic-commons' at the moment.